File: backend/seed_users.py
"""
Seed admin user for GroFresh
"""
import os
from pymongo import MongoClient
from datetime import datetime
from passlib.context import CryptContext

MONGO_URL = os.environ.get('MONGO_URL', 'mongodb://localhost:27017/')
client = MongoClient(MONGO_URL)
db = client['grofresh']

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")

# Create admin user
admin_email = "admin@grofresh.com"
admin_password = "admin123"

# Check if admin exists
existing_admin = db.users.find_one({"email": admin_email})

if existing_admin:
    print(f"⚠️  Admin user already exists: {admin_email}")
else:
    admin_user = {
        "email": admin_email,
        "full_name": "Admin User",
        "hashed_password": pwd_context.hash(admin_password),
        "role": "admin",
        "created_at": datetime.now()
    }
    
    result = db.users.insert_one(admin_user)
    print(f"✅ Admin user created!")
    print(f"   Email: {admin_email}")
    print(f"   Password: {admin_password}")
    print(f"   Role: admin")

# Create regular test user
user_email = "user@grofresh.com"
user_password = "user123"

existing_user = db.users.find_one({"email": user_email})

if existing_user:
    print(f"⚠️  Test user already exists: {user_email}")
else:
    test_user = {
        "email": user_email,
        "full_name": "Test User",
        "hashed_password": pwd_context.hash(user_password),
        "role": "user",
        "created_at": datetime.now()
    }
    
    result = db.users.insert_one(test_user)
    print(f"✅ Test user created!")
    print(f"   Email: {user_email}")
    print(f"   Password: {user_password}")
    print(f"   Role: user")

print("\n✅ User seeding complete!")
client.close()

File: backend-node/src/seeds/seed-users.ts
/**
 * Seed script — Admin + test user
 * Mirrors backend/seed_users.py exactly.
 *
 * Usage:
 *   npx ts-node src/seeds/seed-users.ts
 */

import dotenv from 'dotenv';
dotenv.config();

import bcrypt from 'bcryptjs';
import mongoose from 'mongoose';
import { env } from '../config/env';
import { UserModel } from '../modules/auth/auth.model';

async function seedUsers(): Promise<void> {
  const mongoUri = `${env.MONGO_URL.replace(/\/$/, '')}/${env.MONGO_DB_NAME}`;
  await mongoose.connect(mongoUri);

  const users = [
    {
      email: 'admin@grofresh.com',
      full_name: 'Admin User',
      password: 'admin123',
      role: 'admin' as const,
    },
    {
      email: 'user@grofresh.com',
      full_name: 'Test User',
      password: 'user123',
      role: 'user' as const,
    },
  ];

  for (const u of users) {
    const existing = await UserModel.findOne({ email: u.email });

    if (existing) {
      console.log(`⚠️  User already exists: ${u.email}`);
      continue;
    }

    const hashed_password = await bcrypt.hash(u.password, 10);

    await UserModel.create({
      email: u.email,
      full_name: u.full_name,
      hashed_password,
      role: u.role,
      created_at: new Date(),
    });

    console.log(`✅ ${u.role === 'admin' ? 'Admin' : 'Test'} user created!`);
    console.log(`   Email:    ${u.email}`);
    console.log(`   Password: ${u.password}`);
    console.log(`   Role:     ${u.role}`);
  }

  await mongoose.connection.close();
}

seedUsers().catch((err: unknown) => {
  console.error('Seed failed:', err);
  process.exit(1);
});
